app/jobs/reports/month_helper.rb

```diff
@@ -8,19 +8,24 @@ module MonthHelper
     # @example
     #   months(Date.new(2021, 3, 15)..Date.new(2021, 5, 15))
     #   => [
-    #     Date.new(2021, 3, 15)..Date.new(2021, 3, 31),
-    #     Date.new(2021, 4, 1)..Date.new(2021, 4, 30),
-    #     Date.new(2021, 5, 1)..Date.new(2021, 5, 15),
+    #     Time.new(2021, 3, 15, 0, 0, 0)..Time.new(2021, 3, 31, 23, 59, 59),
+    #     Time.new(2021, 4, 1, 0, 0, 0)..Time.new(2021, 4, 30, 23, 59, 59),
+    #     Time.new(2021, 5, 1, 0, 0, 0)..Time.new(2021, 5, 14, 23, 59, 59),
     #   ]
     # @param [Range<Date>] date_range
-    # @return [Array<Range<Date>>]
+    # @return [Array<Range<Time>>]
     def months(date_range)
+      time_range = Range.new(
+        date_range.begin.in_time_zone('UTC').beginning_of_day,
+        (date_range.end - 1).in_time_zone('UTC').end_of_day,
```

Comment:
TBD: should this check that the beginning/end overlap? (getting confirmation from billing team) that way we don't subtract one if the IAA switches to say, 1/15...1/14

cc @18F/identity-ursula for ideas on how we can make sure our data entry process for these stays consistent

Reply:
@zachmargolis could you clarify what the consideration is here? Thanks!

Reply:
ok so, typically iaa_start/end dates are entered as `2021-1-15` to `2022-1-15`. So when we turn those into timestamps, we want the beginning/end of day. So what this does is it turns those into `2021-1-15 @ 00:00:00` and `2022-01-14 @ 23:59:59` because that's the full year.

The problem would be if we switched to recording dates as `2021-1-15` to `2022-1-14` (note one day off), and if we did that, this `- 1.day` would break that assumption

Reply:
So another option would be to special-case these overlapping start/end dates so something like:

```ruby
if start.month == end.month && start.day == end.day
  # overlapping days, we should -1
else
  # not overlapping, don't  -1
end
```

Reply:
I don't think that's accurate, IAAs generally are from, say `2021-01-15` to `2022-01-14`, at least in my experience, where the agreement expires at `2022-01-14 23:59:59 ET`. Have we been subtracting a day the whole time?

Reply:
we have not been subtracting a day, but I thought we coded them to overlap more.

I could only find one current example of overlapping days, looks like most are like you're saying:

https://github.com/18F/identity-idp-config/blob/b52a63558fdeff076405f2faddf5f8a730b06e7b/iaa_orders.yml#L57-L58

So I think the correct thing for me to do now is remove this `-1`

Reply:
updated in 9cda5384761bea135fbb72904f7e07a1e8bad89c

Reply:
FWIW that IAA was an 11-month IAA (started in September, ended in August) and it ended early because it was subsumed by another agreement.

Reply:
But yes, agree that we should not have the -1 😄. Thanks for checking!